src/pages/App.tsx

```diff
@@ -72,10 +73,25 @@ const App = () => (
                             />
                             <Route
                               exact
-                              path="/market/:assetId"
-                              component={
-                                process.env.REACT_APP_RUN_V2 ? MarketDetails : MarketDetailsV1
-                              }
+                              path="/market/:vTokenId"
+                              render={props => {
+                                // Redirect to market page if vTokenId is
+                                // invalid
+                                if (
+                                  !Object.prototype.hasOwnProperty.call(
+                                    VBEP_TOKENS,
+                                    props.match.params.vTokenId,
+                                  )
+                                ) {
+                                  return <Redirect to="/market" />;
```

Comment:
I think Redirect logic is better organized in the MarketDetails page

Reply:
I don't think that's ideal. The router should be responsible of handling redirections, while the page should only care about rendering its content

Reply:
I don't think that is always going to be possible. There will be business logic required at times where it won't make sense to lift that logic/ state to the router. This page is redirecting because it can't render its content because it received an invalid route but we may redirect for many other reasons from with components/ side effect containers.

Additionally, If we include redirect logic inline for every page in the router this component will unwieldily. At that point we'd like want a special component to redirect. 

I don't feel strongly about this unless this component becomes messy

Reply:
Ok I see, your point about this file becoming messy makes sense to me.
I've made the update; the Market Details page now handle the redirection.
I also moved all the route paths to a global `Path` enum so that they are all defined from a single source of truth.